solr/core/src/java/org/apache/solr/search/SolrIndexSearcher.java

```diff
@@ -1660,9 +1656,7 @@ private DocSet getDocListAndSetNC(QueryResult qr, QueryCommand cmd) throws IOExc
 
     ProcessedFilter pf = getProcessedFilter(cmd.getFilter(), cmd.getFilterList());
     Query query = QueryUtils.makeQueryable(cmd.getQuery());
-    if (pf.filter != null) {
-      query = new BooleanQuery.Builder().add(query, Occur.MUST).add(pf.filter, Occur.FILTER).build();
-    }
+    query = QueryUtils.combineQueryAndFilter(query, pf.filter);
```

Comment:
```suggestion
    final Query query = QueryUtils.combineQueryAndFilter(QueryUtils.makeQueryable(cmd.getQuery()), pf.filter);
```
could provide clarity w.r.t. `query` not subsequently changing though the `QueryUtils...QueryUtils` bit is a bit ugly perhaps.

Reply:
I was thinking the same thing but did not act.  I will now :-)